clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/LeaderFollowerStoreIngestionTask.java

```diff
@@ -3506,6 +3538,68 @@ protected Set<String> maybeSendIngestionHeartbeat() {
     return failedPartitions;
   }
 
+  /**
+   *  Resubscribe operation by passing new version role and partition role to {@link AggKafkaConsumerService}. The action
+   *  for leader and follower replica will be handled differently.
+   */
+  private void resubscribe(
+      PartitionConsumptionState partitionConsumptionState,
+      TopicPartitionReplicaRole.VersionRole versionRoleFromConsumerAction) throws InterruptedException {
```

Comment:
Can we do a shortcuit and skip the resubscribe if version role doesn't change? For example, a store config change like increasing quota will trigger store change callback; in this case, SIT's version role doesn't change, and there is no point doing resubscription.

Reply:
Actually, we cached the `verisonRole` inside the  `storeVersionRoleChangedListener` and compare new role with previous one to see if issuing `VersionRoleChange` Consumer Action.